pkg/sanity/sanity.go

```diff
@@ -156,18 +170,35 @@ func (sc *SanityContext) teardown() {
 	// (https://github.com/kubernetes-csi/csi-test/pull/98).
 }
 
-func createMountTargetLocation(targetPath string) error {
-	fileInfo, err := os.Stat(targetPath)
-	if err != nil && os.IsNotExist(err) {
-		return os.MkdirAll(targetPath, 0755)
-	} else if err != nil {
-		return err
+func createMountTargetLocation(targetPath string, customCreateDir func(string) (string, error)) (string, error) {
+
+	// Return the target path if empty.
+	if len(targetPath) < 0 {
+		return targetPath, nil
 	}
-	if !fileInfo.IsDir() {
-		return fmt.Errorf("Target location %s is not a directory", targetPath)
+
+	var newTargetPath string
+
+	if customCreateDir != nil {
+		newpath, err := customCreateDir(targetPath)
+		if err != nil {
+			return "", err
+		}
+		newTargetPath = newpath
+	} else {
+		fileInfo, err := os.Stat(targetPath)
+		if err != nil && os.IsNotExist(err) {
+			return "", os.MkdirAll(targetPath, 0755)
```

Comment:
Here you return an empty string also when `os.MkdirAll` succeeds. That can't be right? I wonder why it isn't breaking the Travis CI check. Ah, probably because the mock driver doesn't actually use the directories and doesn't check whether they exist (unverified). Might be worth adding such a check. Perhaps add such a check as part of the PR, verify "make test" fails, then fix this code?

Overall I find this version more readable:
```
		fileInfo, err := os.Stat(targetPath)
		if err != nil {
                     if !os.IsNotExist(err) {
                        return "", err
                     }
	             if err := os.MkdirAll(targetPath, 0755); err != nil {
                          return "", err
                     }
                }
		if !fileInfo.IsDir() {
			return "", fmt.Errorf("Target location %s is not a directory", targetPath)
		}

Reply:
oops! thanks for noticing that.
I've added new checks in NodePublishVolume and NodeStageVolume to fail if the target paths don't exist. And adopted the above code snippet, this is actually better.